File: conexiones/conexionPost.go
package main

import (
	"database/sql"
	"fmt"
	_ "github.com/lib/pq"
)

func main() {
	// Cadena de conexión: Ajusta host, usuario, contraseña y nombre de la base de datos
	connStr := "host=localhost user=postgres password=12345 dbname=tienda sslmode=disable"
	db, err := sql.Open("postgres", connStr)
	if err != nil {
		fmt.Println("Error al abrir la conexión:", err)
		return
	}
	defer db.Close()

	err = db.Ping()
	if err != nil {
		fmt.Println("Error de conexión:", err)
		return
	}
	fmt.Println("Conexión exitosa a PostgreSQL")
}

File: conexiones/conexionPost.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class conexionPost {
    public static void main(String[] args) {
        String url = "jdbc:postgresql://localhost:5432/"; // Ajusta la URL y el nombre de la base de datos si es necesario
        String user = "postgres";
        String password = "12345";
        
        try {
            Connection conn = DriverManager.getConnection(url, user, password);
            System.out.println("Conexion exitosa");
            conn.close();
        } catch (SQLException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}
